packages/react-core/src/filters/applyFiltersToTileContent.js

```diff
@@ -0,0 +1,31 @@
+import { buildBinaryFeatureFilter } from './Filter';
+
+export function applyFiltersToTileContent(tileContent, filters) {
+  if (!filters || !Object.keys(filters).length || !tileContent) {
+    return tileContent;
+  }
+
+  const filterFn = buildBinaryFeatureFilter({ filters });
+
+  const applyFiltersFn = (data) => (_, idx) =>
+    (!data.attributes || data.attributes.getFilterValue.value[idx]) &&
+    filterFn(idx, data);
+
+  return {
+    ...tileContent,
+    ...['points', 'lines', 'polygons'].reduce((acc, key) => {
+      acc[key] = {
+        ...tileContent[key],
+        attributes: {
+          getFilterValue: {
+            value: tileContent[key].featureIds.value.map(
```

Comment:
Instead of use `tileContent[key].featureIds` we could use `primitivePolygonIndices` or `pathIndices` or `pointIndices`. If we use `featureIds` we're repeating calculation for lines and polygons

Reply:
Good catch 👌  I'll change it.

Reply:
Done